jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/GettingStartedAuthUtils.kt

```diff
@@ -156,6 +157,91 @@ fun requestCredentialsForCodeWhisperer(
     return isAuthenticationSuccessful
 }
 
+fun requestCredentialsForCodeCatalyst(
+    project: Project,
+    popupBuilderIdTab: Boolean = true,
+    initialConnectionCount: Int = getConnectionCount(),
+    initialAuthConnections: String = getEnabledConnections(
+        project
+    ),
+    isFirstInstance: Boolean = false,
+    connectionInitiatedFromExplorer: Boolean = false
+): Boolean {
+    val authenticationDialog = SetupAuthenticationDialog(
+        project,
+        state = SetupAuthenticationDialogState().also {
+            if (popupBuilderIdTab) {
+                it.selectedTab.set(SetupAuthenticationTabs.BUILDER_ID)
+            }
+        },
+        tabSettings = mapOf(
+            SetupAuthenticationTabs.IDENTITY_CENTER to AuthenticationTabSettings(
+                disabled = false,
+                notice = SetupAuthenticationNotice(
+                    SetupAuthenticationNotice.NoticeType.WARNING,
+                    message("gettingstarted.setup.codecatalyst.use_builder_id"),
+                    "https://docs.aws.amazon.com/codewhisperer/latest/userguide/codewhisperer-auth.html"
+                )
+            ),
+            SetupAuthenticationTabs.BUILDER_ID to AuthenticationTabSettings(
+                disabled = false,
+                notice = SetupAuthenticationNotice(
+                    SetupAuthenticationNotice.NoticeType.WARNING,
+                    message("gettingstarted.setup.codecatalyst.use_identity_center"),
+                    "https://docs.aws.amazon.com/codewhisperer/latest/userguide/codewhisperer-auth.html"
+                )
+            ),
+            SetupAuthenticationTabs.IAM_LONG_LIVED to AuthenticationTabSettings(
+                disabled = true,
+                notice = SetupAuthenticationNotice(
+                    SetupAuthenticationNotice.NoticeType.ERROR,
+                    message("gettingstarted.setup.codecatalyst.no_iam"),
+                    "https://docs.aws.amazon.com/codewhisperer/latest/userguide/codewhisperer-auth.html"
+                )
+            )
+        ),
+        scopes = CODECATALYST_SCOPES,
+        promptForIdcPermissionSet = false,
+        sourceOfEntry = SourceOfEntry.CODECATALYST,
+        featureId = FeatureId.Codecatalyst,
+        isFirstInstance = isFirstInstance,
+        connectionInitiatedFromExplorer = connectionInitiatedFromExplorer
+    )
+    val isAuthenticationSuccessful = authenticationDialog.showAndGet()
+    if (isAuthenticationSuccessful) {
+        AuthTelemetry.addConnection(
```

Comment:
For Telemetry, we will also need to update the function: `getEnabledConnectionsForTelemetry` to reflect whether the enabled connections are Idc or Builder Id